dojo/tools/sarif/parser.py

```diff
@@ -149,10 +138,92 @@ def cve_try(val):
         return None
 
 
+def get_title(result, rule):
+    title = get_message_from_multiformatMessageString(result['message'], rule)
+    if rule is not None:
+        if title is None and 'shortDescription' in rule:
+            title = get_message_from_multiformatMessageString(rule['shortDescription'], rule)
+        elif title is None and 'fullDescription' in rule:
+            title = get_message_from_multiformatMessageString(rule['fullDescription'], rule)
+        elif title is None and 'name' in rule:
+            title = rule['name']
+        elif title is None:
+            title = rule['id']
+
+    if title is None:
+        title = ''
```

Comment:
Not a fan of producing fake blank data.

Reply:
Me neither, but title cannot be None. Shall we throw an error, when there is no title?

Reply:
I think maybe. what bother me is that the blank title could introduce some confusion for people (is it a bug (no title)?) and the code of DD (default de-duplication algorithm rely on the title I think)

Reply:
Then I will change it to raise a `ValueError`. A lot of fields are tried to get a title. When all this fails, the report might be unusable anyway.

Reply:
Done

Reply:
awesome